Feat/statistics main page

dilkusha27 opened the pull request:
## #️⃣ 연관된 이슈

> 이 PR과 관련된 이슈 번호를 적고, 해당 이슈를 닫으세요. 예: closes #이슈번호

## #️⃣ 작업 내용

> 이번 PR에서 작업한 내용을 간략히 설명해주세요. (이미지 첨부 가능)

## #️⃣ 테스트 결과

> 코드 변경에 대해 테스트를 수행한 결과를 요약해주세요. 예: 모든 테스트 통과 여부, 새로 작성한 테스트 케이스 등

## #️⃣ 변경 사항 체크리스트

- [ ] 코드에 영향이 있는 모든 부분에 대한 테스트를 작성하고 실행했나요?
- [ ] 필요한 경우, 문서를 작성하거나 수정했나요?
- [ ] 코드 컨벤션에 따라 코드를 작성했나요?
- [ ] 본 PR에서 발생할 수 있는 모든 의존성 문제가 해결되었나요?

## #️⃣ 스크린샷 (선택)

> 관련된 스크린샷이 있다면 여기에 첨부해주세요.

## #️⃣ 리뷰 요구사항 (선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요. 
> 예시: 이 부분의 코드가 잘 작동하는지 테스트 해주실 수 있나요?

## 📎 참고 자료 (선택)

> 관련 문서, 스크린샷, 또는 예시 등이 있다면 여기에 첨부해주세요.

dilkusha27 committed:
feat: Refresh Token을 이용한 Access Token 갱신 기능 구현

- Axios Interceptors 이용하여 API 호출 시 요청, 응답 전 자동으로 Access Token 추가
- 401 응답 발생 시 Refresh Token을 이용해 Access Token 자동 갱신
- 갱신 실패 시 localStorage 초기화 후 로그인 페이지로 리다이렉트
- 401 외 에러는 그대로 반환

dilkusha27 committed:
feat: Header에서 사용자 정보 표시

dilkusha27 committed:
refactor: 캘린더 시간대 서울 기준으로 변경

dilkusha27 committed:
fix: 컴포넌트 명칭 수정

- 대소문자 차이로 인한 redirectHandler.tsx 페이지 컴포넌트 오류 발생
- 컴포넌트 명 및 export 문 PascalCase로 변경

dilkusha27 committed:
fix: 코드 수정 중 타입 정의 에러로 인해 이전 상태로 되돌림

dilkusha27 committed:
feat: 메인 페이지 통계 데이터 패치 및 렌더링

wkddnjswl7 commented on src/services/api.tsx:
이 api 파일은 refresh token에 대한 내용을 담고 있는 파일일까요?

dilkusha27 replied:
네 맞습니다!

wkddnjswl7 replied:
그러면 파일명을 한번에 알기 쉽게 reFreshTokenApi로 하는건 어떠신가요??
순간 제가 Api 잘못만든 줄 알고 다시 만드신 줄 알았어요 ...

dilkusha27 replied:
아 그럴 수도 있었겠네요... 명칭 변경하고 Auth 파일에 넣어뒀습니다!

wkddnjswl7 replied:
확인했습니다 ! 병합 진행하겠습니다 감사합니다 :)

wkddnjswl7 reviewed (commented):
리뷰 확인 부탁드립니다!

dilkusha27 committed:
Merge branch 'dev' into feat/statistics-main-page

dilkusha27 committed:
refactor: 일기 생성 페이지 이동 및 선택 날짜 데이터 전송

wkddnjswl7 reviewed (commented):
리뷰 재등록했습니다 ! 확인부탁드립니다:)

dilkusha27 committed:
refactor: CreateDiary 페이지에서 이전 캘린더에서 선택한 날짜를 url에서 받아오도록 초기값 설정

dilkusha27 committed:
refactor: 불필요하게 전달되는 props 제거 및 prettier 적용으로 인한 코드 변경 (내용 변화 없음)

dilkusha27 committed:
refactor: 토큰 갱신 API 모듈 파일 명칭 변경

dilkusha27 committed:
refactor: 파일명 및 위치 변경

dilkusha27 committed:
refactor: 기능과 직접적인 관련이 적은 코드 수정

- console에 출력되는 내용 구분하기 쉽게 수정
- 변수 타입 실제 데이터 형태에 맞춰 수정
- 하위 컴포넌트에 전달하는 props 수정
